File: 2023/string-230417_0512/2002.py
import sys
input = lambda: sys.stdin.readline().rstrip()

N = int(input())
enter = { input(): i for i in range(N) }
out = [ enter[input()] for _ in range(N) ]

ret = 0
for i in range(N):
	for j in range(i + 1, N):
		if out[i] > out[j]:
			ret += 1
			break
print(ret)

File: boj/2023/0-str-230417_0512/2002.cpp
#include <cstdio>
#include <map>
#include <string>

using namespace std;

int main()
{
    int N;
    scanf("%d", &N);

    char s[10];

    map<string, int> enter;
    for (int i = 0; i < N; i++) {
        scanf("%s", s);
        enter[s] = i;
    }

    int out[N];
    for (int i = 0; i < N; i++) {
        scanf("%s", s);
        out[i] = enter[s];
    }
    
    int ret = 0;
    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            if (out[i] > out[j]) {
                ret++;
                break;
            }
        }
    }
    printf("%d", ret);
}